docs/openapi.yaml

```diff
@@ -2,12 +2,72 @@ openapi: 3.0.0
 info:
   title: ArNS Microservice
   version: 0.1.0
-  description: A koa microservice that provides API interface for fetching and retrieving ArNS related Smartweave contracts.
-
+description: A koa microservice that provides API interface for fetching and retrieving ArNS related Smartweave contracts.
 servers:
   - url: '/v1'
 
 components:
+  parameters:
+    contractTxId:
+      in: path
+      name: contractTxId
+      required: true
+      description: Transaction ID of the contract.
+      schema:
+        type: string
+        example: 'bLAgYxAdX2Ry-nt6aH2ixgvJXbpsEYm28NgJgyqfs-U'
+    walletAddress:
+      in: path
+      name: walletAddress
+      required: true
+      description: The wallet address.
+      schema:
+        type: string
+        example: 'QGWqtJdLLgm2ehFWiiPzMaoFLD50CnGuzZIPEdoDRGQ'
+    name:
+      in: path
+      name: name
+      required: true
+      description: The name of the record to return.
+      schema:
+        type: string
+        example: 'ario'
+    blockHeight:
+      name: blockHeight
+      in: query
+      required: false
+      description: Evaluate the contract up to a specific block height. Only applicable if sortKey is not provided.
```

Comment:
Nice. Too bad OpenAPI doesn't make it easier to specify mutual exclusivity among query parameters.

Reply:
+1